test/models/API1811/DocumentDetailsSpec.scala

```diff
@@ -50,24 +50,65 @@ class DocumentDetailsSpec extends SpecBase {
-      "there is a posted penalty, so that no penalty details are provided" in {
-        val model = fullDocumentDetails.copy(penaltyType = Some("LPP1"), penaltyStatus = Some("POSTED"))
-        val expectedOutput = Json.obj(
-          "chargeType" -> "VAT Return Debit Charge",
-          "periodKey" -> "13RL",
-          "taxPeriodFrom" -> "2017-04-06",
-          "taxPeriodTo" -> "2018-04-05",
-          "chargeReference" -> "XM002610011594",
-          "mainTransaction" -> "4700",
-          "subTransaction" -> "1174",
-          "originalAmount" -> 45552768.79,
-          "outstandingAmount" -> 297873.46,
-          "clearedAmount" -> 45254895.33,
-          "items" -> Json.arr(fullLineItemDetailsOutputJson),
-          "accruingInterestAmount" -> 0.23,
-          "interestRate" -> 3
-        )
-        Json.toJson(model) shouldBe expectedOutput
+      "getAccruingPenalty" should {
+
+        "only write the accruing charge when there is an accruing and a posted penalty" in {
+
+          val model = fullDocumentDetails.copy(
+            documentPenaltyTotals = Some(Seq(documentPenaltyTotalsPosted, documentPenaltyTotals)))
+          val expectedOutput: JsObject = Json.obj(
+            "chargeType" -> "VAT Return Debit Charge",
+            "periodKey" -> "13RL",
+            "taxPeriodFrom" -> "2017-04-06",
+            "taxPeriodTo" -> "2018-04-05",
+            "chargeReference" -> "XM002610011594",
+            "mainTransaction" -> "4700",
+            "subTransaction" -> "1174",
+            "originalAmount" -> 45552768.79,
+            "outstandingAmount" -> 297873.46,
+            "clearedAmount" -> 45254895.33,
+            "items" -> Json.arr(fullLineItemDetailsOutputJson),
+            "accruingInterestAmount" -> 0.23,
+            "interestRate" -> 3,
+            "accruingPenaltyAmount" -> 10.01,
+            "penaltyType" -> "LPP1"
+          )
+          Json.toJson(model) shouldBe expectedOutput
+        }
+
+        "return no penalty information as penaltyStatus is not accruing and is posted" in {
+
+          val model = fullDocumentDetails.copy(
+            documentPenaltyTotals = Some(Seq(documentPenaltyTotalsPosted)))
+          val expectedOutput: JsObject = Json.obj(
+            "chargeType" -> "VAT Return Debit Charge",
+            "periodKey" -> "13RL",
+            "taxPeriodFrom" -> "2017-04-06",
+            "taxPeriodTo" -> "2018-04-05",
+            "chargeReference" -> "XM002610011594",
+            "mainTransaction" -> "4700",
+            "subTransaction" -> "1174",
+            "originalAmount" -> 45552768.79,
+            "outstandingAmount" -> 297873.46,
+            "clearedAmount" -> 45254895.33,
+            "items" -> Json.arr(fullLineItemDetailsOutputJson),
+            "accruingInterestAmount" -> 0.23,
+            "interestRate" -> 3
+          )
+          Json.toJson(model) shouldBe expectedOutput
+        }
+
+        "return an empty DocumentPenaltyTotals when PenaltyStatus is Posted" in {
```

Comment:
This and the test below are good, but what happens if document penalty totals is `None`?